### PR TITLE
Approve Add Store Procedures branch

### PR DESCRIPTION
**Pull request by VladPulyak**

### CODE REVIEW

File: DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/TaskStateRepository.cs [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+        public DIMSContext Context { get; }
+
+        public async AsyncTask SetUserTaskAsFail(int userId, int taskId)
+        {
+            await Context.Database.ExecuteSqlRawAsync("execute SetUserTaskAsFail @UserId, @TaskId", userId, taskId);
+        }
+
+        public async AsyncTask SetUserTaskAsSuccess(int userId, int taskId)
+        {
+            await Context.Database.ExecuteSqlRawAsync("execute SetUserTaskAsSuccess @UserId, @TaskId", userId, taskId);
```
Comment (dbogdan2000) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/TaskStateRepository.cs]: It is better to move these methods to UserTaskRepository.

**Review by dbogdan2000 — CHANGES_REQUESTED** [review]

File: DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/TaskStateRepository.cs (hunk 1 above)
Comment (dbogdan2000) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/TaskStateRepository.cs]: @VladPulyak Also you will have too add interface for this repository with these methods

Comment (VladPulyak) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/TaskStateRepository.cs]: Sorry. I didn't notice file name

Comment (VladPulyak) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Repositories/TaskStateRepository.cs]: Fixed

File: DIMS-Core/DIMS-Core.DataAccessLayer/Interfaces/IUserTaskStoreProcedures.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DIMS_Core.DataAccessLayer.Interfaces
+{
+    public interface IUserTaskStoreProcedures
```
Comment (dbogdan2000) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Interfaces/IUserTaskStoreProcedures.cs]: It should be called as IUserTaskRepository and it should inherit IRepository.

**Review by dbogdan2000 — CHANGES_REQUESTED** [review]

**Review by dbogdan2000 — APPROVED** [review]

**Pull request merged**